kie-spring-boot/jbpm-spring-boot/src/test/java/org/jbpm/springboot/RuntimeManagerComponent.java

```diff
@@ -0,0 +1,327 @@
+package org.jbpm.springboot;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import javax.persistence.EntityManagerFactory;
+
+import org.drools.core.audit.WorkingMemoryInMemoryLogger;
+import org.jbpm.process.instance.event.DefaultSignalManagerFactory;
+import org.jbpm.process.instance.impl.DefaultProcessInstanceManagerFactory;
+import org.jbpm.runtime.manager.impl.DefaultRegisterableItemsFactory;
+import org.jbpm.runtime.manager.impl.SimpleRegisterableItemsFactory;
+import org.jbpm.services.task.identity.JBossUserGroupCallbackImpl;
+import org.jbpm.test.util.PoolingDataSource;
+import org.kie.api.event.process.ProcessEventListener;
+import org.kie.api.event.rule.AgendaEventListener;
+import org.kie.api.io.ResourceType;
+import org.kie.api.runtime.manager.Context;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.manager.RuntimeEnvironment;
+import org.kie.api.runtime.manager.RuntimeEnvironmentBuilder;
+import org.kie.api.runtime.manager.RuntimeManager;
+import org.kie.api.runtime.manager.RuntimeManagerFactory;
+import org.kie.api.runtime.manager.audit.AuditService;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemHandler;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.kie.api.task.TaskLifeCycleEventListener;
+import org.kie.api.task.UserGroupCallback;
+import org.kie.internal.io.ResourceFactory;
+import org.springframework.stereotype.Component;
+
+@Component("runtimeManagerComponent")
+public class RuntimeManagerComponent {
```

Comment:
I'd prefer we base spring boot on jbpm services ap rather than runtime manager as it gives much more flexibility.

Reply:
+1 I just did this to show example to have some pre-set-up components users can just inject and use. Ideally we would use services as you mention.